src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -608,81 +608,85 @@ bool GetPeerNodeInfo(CommandHandler * commandHandler, const ConcreteCommandPath
-                        handle.Release();
-                        gOtaSender.ResetState();
-                        return;
-                    }
-                    auto targetNodeId
-                        = handler->GetExchangeContext()->GetSessionHandle()->AsSecureSession()->GetLocalScopedNodeId();
+    auto completionHandler
+        = ^(MTROTASoftwareUpdateProviderClusterQueryImageResponseParams * _Nullable data, NSError * _Nullable error) {
+              [controller
+                  asyncDispatchToMatterQueue:^() {
+                      assertChipStackLockedByCurrentThread();
 
-                    char uriBuffer[kMaxBDXURILen];
-                    MutableCharSpan uri(uriBuffer);
-                    err = bdx::MakeURI(targetNodeId.GetNodeId(), AsCharSpan(data.imageURI), uri);
-                    if (CHIP_NO_ERROR != err) {
-                        LogErrorOnFailure(err);
-                        handler->AddStatus(cachedCommandPath, Protocols::InteractionModel::Status::Failure);
-                        handle.Release();
-                        gOtaSender.ResetState();
-                        return;
-                    }
+                      CommandHandler * handler = EnsureValidState(handle, cachedCommandPath, "QueryImage", data, error);
+                      VerifyOrReturn(handler != nullptr);
 
-                    response.imageURI.SetValue(uri);
-                    handler->AddResponse(cachedCommandPath, response);
-                    handle.Release();
-                    return;
-                }
-                if (!hasUpdate) {
-                    // Send whatever error response our delegate decided on.
-                    handler->AddResponse(cachedCommandPath, response);
-                } else {
-                    // We must have isBDXProtocolSupported false.  Send the corresponding error status.
-                    Commands::QueryImageResponse::Type protocolNotSupportedResponse;
-                    protocolNotSupportedResponse.status
-                        = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusDownloadProtocolNotSupported);
-                    handler->AddResponse(cachedCommandPath, protocolNotSupportedResponse);
-                }
-                handle.Release();
-                gOtaSender.ResetState();
-            }
+                      ChipLogDetail(Controller, "QueryImage: application responded with: %s",
+                          [[data description] cStringUsingEncoding:NSUTF8StringEncoding]);
 
-                          errorHandler:^(NSError *) {
-                              // Not much we can do here
-                          }];
-    };
+                      Commands::QueryImageResponse::Type delegateResponse;
+                      ConvertFromQueryImageResponseParms(data, delegateResponse);
+
+                      // If BDX protocol is not supported, send a response with status - Protocol Not supported and reset state.
```

Comment:
Also, this is not checking "is BDX supported".  It's checking "did the requestor ask us to use the BDX protocol, or some other protocol"?  It's the _other_ protocol that's not supported, by us.

Reply:
correct. will fix the comments

Reply:
done